test/dart/_leak_checker_test.dart

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:leak_tracker/leak_tracker.dart';
+import 'package:leak_tracker/src/_leak_checker.dart';
+import 'package:leak_tracker/src/leak_analysis_model.dart';
+import 'package:test/test.dart';
+
+// Enum-like static classes are ok.
+// ignore: avoid_classes_with_only_static_members
+class _SummaryValues {
+  static void verifyValues() {
+    expect(zero.toMessage(), isNot(nonZero.toMessage()));
+    expect(nonZero.toMessage(), nonZeroCopy.toMessage());
+  }
+
+  static const zero = LeakSummary({});
+
+  static const nonZero = LeakSummary({
+    LeakType.gcedLate: 1,
+    LeakType.notDisposed: 2,
+    LeakType.notGCed: 3,
+  });
+
+  static late final nonZeroCopy =
+      LeakSummary(<LeakType, int>{}..addAll(nonZero.totals));
+}
+
+void main() {
+  late _MockLeakProvider leakProvider;
+
+  late _MockStdoutSink stdout;
+  late _MockDevToolsSink devtools;
+  late _ListenedSink listened;
+
+  const period = Duration(milliseconds: 5);
+  late final doublePeriod = Duration(microseconds: period.inMicroseconds);
+
+  LeakChecker leakChecker({
+    required bool checkPeriodically,
+    required bool hasListener,
+    required bool hasStdout,
+    required bool hasDevtools,
+  }) =>
+      LeakChecker(
+        leakProvider: leakProvider,
+        checkPeriod: checkPeriodically ? period : null,
+        leakListener:
+            hasListener ? (summary) => listened.store.add(summary) : null,
+        stdoutSink: hasStdout ? stdout : null,
+        devToolsSink: hasDevtools ? devtools : null,
+      );
+
+  LeakChecker defaultLeakChecker() => leakChecker(
+        checkPeriodically: true,
+        hasListener: false,
+        hasStdout: true,
+        hasDevtools: true,
+      );
+
+  setUp(() {
+    leakProvider = _MockLeakProvider();
+
+    stdout = _MockStdoutSink();
+    devtools = _MockDevToolsSink();
+    listened = _ListenedSink();
+  });
+
+  test('Mocks emulate production well.', () {
+    _SummaryValues.verifyValues();
```

Comment:
it would be good to pull the contents of this function into the test so it is immediately apparent what is being tested